src/io.c

```diff
@@ -9,27 +9,27 @@
 #include <string.h>
 
 #include "io.h"
+#include "mpool.h"
 
 static const uint32_t mask_lo = 0xffff;
 static const uint32_t mask_hi = ~(0xffff);
 
+static struct mpool *mp;
+
 memory_t *memory_new()
 {
-    memory_t *m = malloc(sizeof(memory_t));
-    memset(m->chunks, 0, sizeof(m->chunks));
+    memory_t *m = calloc(1, sizeof(memory_t));
+    /* sizeof(chunk_t) << 5 satisfies the memory requirement for all test cases
```

Comment:
If `sizeof(chunk_t) << 5` does not fit, there should be some mechanism to extend.